pkg/standalone-utils/contracts/test/MockEulerToken.sol

```diff
@@ -0,0 +1,108 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+// Functionality required is onDeposit it needs to pull ERC20s out
+// of users wallet and transfer into Euler protocol. Similar
+// to on withdrawl. it needs to pull funds from EulerProtocol
+// and send back to users wallet (the underlying token)
+
+import "../relayer/interfaces/IMockEulerProtocol.sol";
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IEulerToken.sol";
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+
+
+contract MockEulerToken is IEulerToken, TestToken {
+    // from Euler docs:
+    // in order to invest an asset to earn interest, you need to `deposit` into an eToken
+
+    uint256 public exchangeRateMultiplier;
+
+    // solhint-disable-next-line var-name-mixedcase
+    address public immutable ASSET;
+    IMockEulerProtocol public immutable EULER_PROTOCOL;
+
+    uint256 private _onlyForMockStorage;
+    address private _mockMsgSender;
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        address asset,
+        IMockEulerProtocol eulerProtocol
+    ) TestToken(name, symbol, decimals) {
+        exchangeRateMultiplier = 1;
+        ASSET = asset;
+        EULER_PROTOCOL = eulerProtocol;
+    }
+
+    // convert eToken Balance to underlying
+    function convertBalanceToUnderlying(uint256 balance) external override view returns (uint256) {
+        return balance * exchangeRateMultiplier;
+    }
+
+    function setExchangeRateMultiplier(uint256 _exchangeRateMultiplier) external {
+        require(_exchangeRateMultiplier < 3, "Cannot set exchangeRateMultiplier bigger 3");
+        exchangeRateMultiplier = _exchangeRateMultiplier;
+    }
+
+    // convert underlying Balance to eToken
+    function convertUnderlyingToBalance(uint256 balance) external override view returns (uint256) {
+        return balance * (1/exchangeRateMultiplier);
+    }
+
+    function underlyingAsset() external override view returns (address) {
+        return ASSET;
+    }
+
+    function deposit(uint256 subAccountId, uint256 amount) external override {
+
+        _onlyForMockStorage = subAccountId;
+        // Relayer only uses one account. Meaning no subAccountID required
+        // is set to 0 in the Relayer
+        // Transfer underlying tokens from sender to the Euler pool, and increase account's eTokens
+
+        // assumes Euler protocol has approval to move `amount` from users wallet
+        // mints MockEulerTokens to msg.sender (which is the relayer)
+        // the relayer has earlier used `_pullToken` to ensure it is the contract
+        // which is calling `deposit` on an eToken.
+
+        _mockMsgSender = msg.sender;
+
+        EULER_PROTOCOL.requestUnderlyingFromRelayer(address(ASSET), amount, _mockMsgSender); 
```

Comment:
Here, I'd just pass in `msg.sender` directly instead of writing it to a variable. Also, it's not necessary to cast `ASSET` since it is already of type `address`.
```suggestion
        EULER_PROTOCOL.requestUnderlyingFromRelayer(ASSET, amount, msg.sender); 
```